### PR TITLE
Circleci

### PR DESCRIPTION
**Pull request by AaronPlave**
Would like to merge this into master instead of dev so we can get a baseline CI setup. This PR doesn't include Coveralls work or branch matching for *-feature type branch deployments. 

Also this PR definitely needs to be squashed.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                # command: npm run test
+
+            # run build
+            - run:
+                name: Run build
+                command: npm run build
+
+            - add_ssh_keys:
+                fingerprints:
+                    - "2a:ed:6a:f3:34:9d:7a:17:e3:75:74:2d:4f:ab:75:7d"
```
Comment (flynnplatt) [inline .circleci/config.yml]: these look so secretive, but I'm assuming this is more or less a public key?

File: scripts/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+# Remove unneeded files from gh-pages
+shopt -s extglob
+rm -rf !(test-results|dist|assets|branches)
+
+if [ -f ".gitignore"]; then
+    rm .gitignore # remove our .gitignore so we can grab certain items
+fi
+
+# Move test-results output into source branch
+mv test-results branches/$SOURCE_BRANCH/unit-tests
```
Comment (flynnplatt) [inline scripts/deploy.sh]: is 'unit-tests' just like standard or what?

File: .circleci/config.yml (hunk 1 above)
Comment (AaronPlave) [reply .circleci/config.yml]: https://en.wikipedia.org/wiki/Public_key_fingerprint

File: scripts/deploy.sh (hunk 2 above)
Comment (AaronPlave) [reply scripts/deploy.sh]: Nope, that's just the way it was in the old script, wondering if it was done to skirt around that .gitignore issue. Can rename to test-results? Test/coverage stuff will also get another pass when I start doing Coveralls work + work on hooking test results into CircleCI reports.

File: .circleci/config.yml (hunk 1 above)
Comment (flynnplatt) [reply .circleci/config.yml]: right, just making sure

File: scripts/deploy.sh (hunk 2 above)
Comment (flynnplatt) [reply scripts/deploy.sh]: I would just as soon keep it named 'test-results' since that's what's produced in the build

Comment (AaronPlave) [reply scripts/deploy.sh]: But I think it might be best to do that Coveralls stuff outside of this PR.

Comment (flynnplatt) [reply scripts/deploy.sh]: to name it 'unit-tests'?

Comment (flynnplatt) [reply scripts/deploy.sh]: I mean the .gitignore file is gone so there shouldn't be a commit issue, so why rename to 'unit-tests'?

Comment (AaronPlave) [reply scripts/deploy.sh]: Ok just remembered one reason – if you take a look at https://github.com/nasa/common-mapping-client/tree/gh-pages/branches/circleci/unit-tests you'll see we're including both test results and coverage in there so "unit-tests" was just some name to group those two things together.

**Pull request merged**